### PR TITLE
ci(typescript): add TypeScript to replace Flow

### LINKED ISSUE #56: Convert from Flow to TypeScript
## Description
Per the discussion in #52, we have decided to move from Flow types to TypeScript to follow industry best practices. 

## What's the benefit of this feature?
1. Moves us in line with current industry standards in Open Source repositories. Flow usage is waning as TS usage is waxing
2. Better integration with text editors than Flow
    - this repo works in large part with objects, and having enhanced editor support for working with those objects (auto-filling properties for example) is a win
 3. Helps auto-document code 
 4. Many other TS benefits...

## Possible Implementation (optional)
This can happen in a few parts. Here are the general broad strokes:
- [ ] rename `.js` files to `.ts`
- [ ] uninstall flow-bin
- [ ] install typescript as a dev dep and initialize typescript
- [ ] convert flow types to typescript types (consider looking at our [definitelytyped](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f7ec78508c6797e42f87a4390735bc2c650a1bfd/types/git-parse/index.d.ts) entry for inspiration)
      - Could also try using this [conversion project](https://github.com/Khan/flow-to-ts)
- [ ] Add TS documentation to the readme and any convenience scripts to the package json
- [ ] stretch goal: add TS checking to our CI

### PR DESCRIPTION
**Pull request by kv979w**
Resolves #56

## Proposed Changes
Describe the big picture of your changes here to communicate to the maintainers the "why" behind this pull request. If it fixes a bug or resolves a feature request, be sure to [link to that issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)

## Types of changes
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Test Plan and Documentation

Tested changes locally

## Further comments

New dist folder contents after running `yarn build`:
Screen Shot 2021-11-16 at 11 27 41 AM

### CODE REVIEW

**Comment by Cooperbuilt** [comment]
Nice work! Code looks good to me, CI is failing at the lint and check typescript steps, but once that is cleared up we are good to go

**Comment by Cooperbuilt** [comment]
@GaryPWhite did you want to knock out the node 12 downgrade before this for any reason?

**Comment by GaryPWhite** [comment]
@Cooperbuilt not particularly! I did just finish up the PR and am happy to merge it before we do this.

File: .github/workflows/pr-checks.yml [hunk 1]
```diff
@@ -61,3 +61,19 @@ jobs:
...
+  typescript:
+    name: Check TypeScript
+    runs-on: ubuntu-latest
+    needs: install-node-modules
+
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 14.x
```
Comment (Cooperbuilt) [inline .github/workflows/pr-checks.yml:74]: Should this be 12?

**Review by Cooperbuilt — APPROVED** [review]
one comment, otherwise lgtm

File: .github/workflows/pr-checks.yml (hunk 1 above)
Comment (KalleV) [reply .github/workflows/pr-checks.yml]: Thanks for noticing - I downgraded to 12x

Comment (Cooperbuilt) [reply .github/workflows/pr-checks.yml]: Sweet, thanks!

**Pull request merged**